src/ol/PluggableMap.js

```diff
@@ -524,6 +555,14 @@ class PluggableMap extends BaseObject {
     layers.push(layer);
   }
 
+  /**
+   * @param {import("./layer/Group.js").GroupEvent} event The layer add event.
+   * @private
+   */
+  handleLayerAdd_(event) {
+    setLayerMapProperty(event.layer, this);
+  }
```

Comment:
We could dispatch an `addlayer` event here on the map if we wanted others to be able to leverage this event (and `removelayer` below).  I don't plan on doing that as part of this change.